src/components/Images.vue

```diff
@@ -112,6 +112,11 @@ export default {
 				return this.src
 			}
 
+			// Check for app provided preview path
+			if (this.$attrs.appPreviewPath) {
+				return this.$attrs.appPreviewPath
+			}
```

Comment:
Hey!

Thanks for this! I think we should also move to this direction. Let's deprecate the `hasPreview`, and use a dedicated prop like `previewUrl`. If there is this prop defined, then we assume there is a valid preview and use it.

Can you make the change and rename it to  `previewUrl` ?

Reply:
Hi @skjnldsv,

Yes, I will make the suggested change. Regarding deprecation of `hasPreview` - if I see correctly there is quite a lot of code around (not only Photos app, but also Talk/spreed) that seem to use `hasPreview`.

Do I understand correctly that we should still leave handling of the `hasPreview` property for compatibility? And just mark as deprecated in the API?

BTW - I am still struggling with Vue and can hardly follow the flow from the top-level `OCA.Viewer.open` call in Photos down to relevant pieces of the Viewer app. I will appreciate any guidance/suggestions.

Reply:
> Do I understand correctly that we should still leave handling of the `hasPreview` property for compatibility? And just mark as deprecated in the API?

yes :)
The deprecation can just be done in the Readme
Btw, this should mostly be done in https://github.com/nextcloud/viewer/blob/84a208915cc82da192a90743a9aecaa17e667385/src/mixins/PreviewUrl.js#L75, not Images.vue.

Reply:
Hi @skjnldsv,

I have implemented handling of the new property `previewUrl` in https://github.com/nextcloud/viewer/pull/1525/commits/d216abe2e16805802637c907a3e22ebe742d27cc. Please check if this looks sane.

I have reverted changes in `Images.vue` as suggested. This has one side effect - the `hasPreview` property must be set to `true` - otherwise the code does not even get to the `previewPath` calculation, but falls back to use the `source` property here:
https://github.com/nextcloud/viewer/blob/d216abe2e16805802637c907a3e22ebe742d27cc/src/components/Images.vue#L118

Seems that a change similar to the original one in `Images.vue` would be necessary if we would like to check for `previewUrl` before checking `hasPreview`.

I have not yet updated Readme - until we confirm the details of the change.

PS. On the second thought, I am not sure if the `hasPreview` property should be deprecated. This provides a nice functionality - easier to use in the other applications than the `previewUrl`.

Reply:
How about adding ` && !this.previewUrl` to this condition:

https://github.com/nextcloud/viewer/blob/d216abe2e16805802637c907a3e22ebe742d27cc/src/components/Images.vue#L117

Reply:
> PS. On the second thought, I am not sure if the hasPreview property should be deprecated. This provides a nice functionality - easier to use in the other applications than the previewUrl.

Could you share some example ?

Reply:
> How about adding ` && !this.previewUrl` to this condition:
> 
> https://github.com/nextcloud/viewer/blob/d216abe2e16805802637c907a3e22ebe742d27cc/src/components/Images.vue#L117

I like this idea very much. This enables setting `hasPreview` to `false` in Photos - so that it works using `source` with released/older versions of Viewer.

Change commited in: https://github.com/nextcloud/viewer/pull/1525/commits/4a080e4257a49079e23091ac86640b7d49326844

Reply:
> > PS. On the second thought, I am not sure if the hasPreview property should be deprecated. This provides a nice functionality - easier to use in the other applications than the previewUrl.
> 
> Could you share some example ?

This is used in Photos for regular albums (not public), also in Talk here:
https://github.com/nextcloud/spreed/blob/b4323721edfc90b8a31c9e0bed1285dd9c5146e4/src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue#L495

Reply:
@starypatyk deprecated != removed.
Apps should learn to migrate to `previewUrl` instead of `haspreview`, they can't know if the documentation doesn't specify the deprecation (jsdoc AND readme)

Reply:
As a quick statement, I will withdraw my initial deprecation idea.
Because we don't want to make every app handle the preview URL generation, we should also keep `hasPreview`